includes/payment-methods/class-wc-stripe-upe-payment-gateway.php

```diff
@@ -495,17 +495,29 @@ public function process_payment( $order_id, $retry = true, $force_save_source =
 		$save_payment_method       = $this->has_subscription( $order_id ) || ! empty( $_POST[ 'wc-' . self::ID . '-new-payment-method' ] ); // phpcs:ignore WordPress.Security.NonceVerification.Missing
 		$selected_upe_payment_type = ! empty( $_POST['wc_stripe_selected_upe_payment_type'] ) ? wc_clean( wp_unslash( $_POST['wc_stripe_selected_upe_payment_type'] ) ) : ''; // phpcs:ignore WordPress.Security.NonceVerification.Missing
 
+		$statement_descriptor                  = ! empty( $this->get_option( 'statement_descriptor' ) ) ? str_replace( "'", '', $this->get_option( 'statement_descriptor' ) ) : '';
+		$short_statement_descriptor            = ! empty( $this->get_option( 'short_statement_descriptor' ) ) ? str_replace( "'", '', $this->get_option( 'short_statement_descriptor' ) ) : '';
+		$is_short_statement_descriptor_enabled = ! empty( $this->get_option( 'is_short_statement_descriptor_enabled' ) ) && 'yes' === $this->get_option( 'is_short_statement_descriptor_enabled' );
+		$descriptor                            = null;
+		if ( 'card' === $selected_upe_payment_type && $is_short_statement_descriptor_enabled && ! empty( $short_statement_descriptor ) ) {
```

Comment:
If the setting ("Add customer order number to the bank statement") is enabled, I feel like we should still include the dynamic bit even if the "Shortened customer bank statement" value is empty. We could truncate it to 10 characters, or just truncate it to whatever length makes it fit with the suffix?

(Applies to non-UPE condition as well.)

Reply:
The validation of the settings is built in a way that it won't let the user save a blank "Shortened customer bank statement" if "Add customer order number to the bank statement" is enabled. I can't see any use case where the proposed behavior would take place. Do you have some specific concern for us to fallback to the regular statement descriptor?

Reply:
Ah, I see – I believe it let me save an empty one, but I can try again.

Reply:
Looks like an unintended consequence of https://github.com/woocommerce/woocommerce-gateway-stripe/pull/2140 since the modified function is used by the validation of both fields?

Reply:
True enough. I'll submit that todo in the settings milestone so we can remember to remove it at some point. I've included the fallback in 7d8e94e.

Reply:
Hm, I'm not too sure about including the whole descriptor as a fallback, since it's not unlikely to be near the limit already, so the order number is likely to be truncated. (Let's say the order number is #673, then the descriptor would be `FULLBANKSTATEMENT* #67`.) Also, this isn't reflected in the Preview, which just shows an empty prefix.

I'm not too concerned about it if this state will be disallowed soon.